root/static/scripts/edit/utility/getUpdatedTrackArtists.js

```diff
@@ -0,0 +1,92 @@
+/*
+ * @flow strict-local
+ * Copyright (C) 2024 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import {reduceArtistCreditNames} from '../../common/immutable-entities.js';
+import {arraysEqual} from '../../common/utility/arrays.js';
+import {cloneArrayDeep} from '../../common/utility/cloneDeep.mjs';
+
+/*
+ * Returns true if the supplied credits include the same underlying artists.
+ */
+const sameArtists = (
+  a: $ReadOnlyArray<ArtistCreditNameT>,
+  b: $ReadOnlyArray<ArtistCreditNameT>,
+) => arraysEqual(a, b,
+  (a, b) => a.artist && b.artist && a.artist.gid === b.artist.gid);
+
+/*
+ * Returns updated artist credits for a track that's part of a release whose
+ * credits are being changed from oldArtists to newArtists.
+ *
+ * If the track's credits should not be changed, the original trackArtists
+ * array is returned.
+ */
+export default function getUpdatedTrackArtists(
+  trackArtists: $ReadOnlyArray<ArtistCreditNameT>,
+  oldArtists: $ReadOnlyArray<ArtistCreditNameT>,
+  newArtists: $ReadOnlyArray<ArtistCreditNameT>,
+): $ReadOnlyArray<ArtistCreditNameT> {
+  const oldArtistsReduced = reduceArtistCreditNames(oldArtists);
```

Comment:
This code previously only reduced these credits once per release, but now it does it once per track. I initially added support for passing in already-reduced credits, but it felt like premature optimization since we still need to reduce the track credits in the next statement.